shell_sort/ShellSort.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import "fmt"
+
+// Worst case time complexity = O(n^2)
+// Best case complexity = O(nlog(n))
+// shellSort returns sorted data
+func shellSort(data []int) []int {
```

Comment:
Will this be similar to call by reference in this case? If yes, i.e. if `data` can be modified in this function, please consider this.